bin/git-team

```diff
@@ -0,0 +1,116 @@
+# limitations under the License.
+# Update repositories - non-Production repos are maintained as
+# shallow copies, to reduce file size and make pulls quicker.
+
+
+# Script to use to save space on target by "not" cloning "shallow" repo with full history and depth=1
+# Parameters:
+#   - `-o` git operation to process - default to `pull`
+#       - For operation other than `pull` shallow repository are deleted and re-cloned to target `branch`
+#       - op can be 'clone', 'checkout', or 'pull';
+#         NB: for 'pull', the branches parameter is not used.
+#   - `-c` repository category to process - team|shallow - default to `shallow`
+#   - `-b` branch to apply operation onto - no effect on `pull` operation - default to `main`
+#   - `-f` fall back branch if `branch` doesn't exists - no effect on `pull` operation - default to `main`
+#   - `-s` shallow repositories to process
+#   - `-p` production repositories to process
+#   - `-p|-s` can be either of:
+#       - "Ensembl" xxx repo name - ex: ensembl-production
+#       - "Git Ensembl config" as define in `./git-ensembl` command tool - ex: production
+#       - "File path" containing a list of repositories to process - one per line
+
+function usage {
+  echo "Usage: $0 [-o git_operation] [-c shallow|team] [-b branch] [-f default_branch] [-t target_dir] [-s shallow_file|repo|config] [-p team_file|repo|team_config]"
+  echo "Example: $0 -o clone -c shallow -s ~/shallow.repo"
+  exit 1
+}
+shallow=()
+prod=()
+while getopts 'o:c:s:p:t:b:f:d' opt; do
+  case ${opt} in
+    o) op=${OPTARG} ;;
+    b) branch=${OPTARG} ;;
+    d) default_branch=${OPTARG} ;;
+    c) category=${OPTARG} ;;
+    t) target_dir=${OPTARG} ;;
+    s) shallow=${OPTARG} ;;
+    p) prod=${OPTARG} ;;
+    *) usage ;;
+  esac
+done
+# repo configuration with file
+if [[ -f "$shallow" ]]; then
+  shallow=(`cat $shallow`)
+fi
+if [[ -f "$prod" ]]; then
+  prod=(`cat $prod`)
+fi
+
+# Ensembl shallow repositories (depth=1)
+branch=${branch:-"main"}
+default_branch=${default_branch:-"main"}
+target_dir=${target_dir:-"${HOME}/git-team"}
+
+if [[ ! -d $target_dir ]]; then
+  mkdir -p $target_dir
+fi
+
+category=${category:-'shallow'}
+op=${op:-'pull'}
+ignore=( "${shallow[@]/#/--ignore_module }" )
+dir=$(pwd)
+cd ${target_dir}
+echo "Processing '${op}' (cat:${category}) from ${PWD} ..."
+echo "=================================================="
+# Using git-tools for pull will update all tags/branches,
+# which we don't want for our shallow copies.
+if [[ $op == 'pull' ]]; then
+  if [[ $category == 'shallow' ]]; then
+    for repo in "${shallow[@]}"; do
+      echo "git ensembl --$op ${repo}"
+      echo "-----------------------------------------"
+      git ensembl --$op ${repo}
+    done
+  else
+    # shellcheck disable=SC2068
+    echo "git ensembl --$op ${ignore[@]} ${prod[@]}"
+    echo "-----------------------------------------"
+    git ensembl --$op ${ignore[@]} ${prod[@]}
+  fi
```

Comment:
Because I'm not using git-tools I'm not sure of what is happening but you seem to be doing line 83..86 what you say you want to avoid in the comment line 79..80

Reply:
Addressed in last commit.